src/hooks/useAutoCompound.ts

```diff
@@ -0,0 +1,226 @@
+import { ISubmittableResult } from '@polkadot/types/types';
+import { BN, u8aToHex } from '@polkadot/util';
+import { decodeAddress, encodeAddress } from '@polkadot/util-crypto';
+import { $api } from 'boot/api';
+import {
+  useCurrentEra,
+  useCustomSignature,
+  useGasPrice,
+  useNetworkInfo,
+  RewardDestination,
+} from 'src/hooks';
+import { TxType } from 'src/hooks/custom-signature/message';
+import { ExtrinsicPayload } from 'src/hooks/helper';
+import { getIndividualClaimTxs, PayloadWithWeight } from 'src/hooks/helper/claim';
+import { getAPI } from 'src/hooks/helper/oakUtils';
+import { getInjector } from 'src/hooks/helper/wallet';
+import { useStore } from 'src/store';
+import { hasExtrinsicFailedEvent } from 'src/store/dapp-staking/actions';
+import { DappCombinedInfo } from 'src/v2/models/DappsStaking';
+import { autoStakingConfig } from 'src/config/autoStaking';
+import { computed, ref, watchEffect } from 'vue';
+
+const MAX_BATCH_WEIGHT = new BN('50000000000');
+// TODO: move to config
+const oakEndpoint = 'wss://rpc.turing-staging.oak.tech';
+
+function getProxyAccount(api: any, address: string, paraId: number) {
+  const decodedAddress = decodeAddress(address); // An Int array presentation of the address’ ss58 public key
+
+  const location = {
```

Comment:
I see that the XCM parameter used is v1. Could you update it to v3? CC: @Dinonard